frontend/src/metabase/query_builder/actions/downloading.ts

```diff
@@ -76,92 +135,101 @@ const getDatasetParams = ({
   visualizationSettings,
 }: DownloadQueryResultsOpts): DownloadQueryResultsParams => {
   const cardId = question.id();
-  const isQuestionInStaticEmbedDashboard =
-    dashcardId != null && cardId != null && token != null;
 
   // Formatting is always enabled for Excel
   const format_rows = enableFormatting && type !== "xlsx" ? "true" : "false";
 
-  if (isQuestionInStaticEmbedDashboard) {
-    return {
-      method: "GET",
-      url: `/api/embed/dashboard/${token}/dashcard/${dashcardId}/card/${cardId}/${type}`,
-      params: Urls.getEncodedUrlSearchParams({ ...params, format_rows }),
-    };
+  const { accessedVia, resourceType: resource } = getDownloadedResourceType({
+    dashboardId,
+    dashcardId,
+    uuid,
+    token,
+    question,
+  });
+
+  // Public links use special endpoints that use uuids instead of ids
+  if (accessedVia === "public-link") {
```

Comment:
The diff is a bit ugly because I also grouped them, first all the public endpoints, then static embed and then the normal ones used by internal/interactive/sdk

Reply:
These conditions are pretty long, I wonder if `ts-pattern` would help with this? Don't bother with it if it makes things worse though.

Reply:
I actually thought that like this they were clearer as I could group them by public-link/static-embed/other.
I did an experiemnt with ts-pattern for `getDownloadedResourceType` but I think it actually looked less readable: https://github.com/metabase/metabase/pull/46132/commits/03615bbea84052998e31d9c7fc6b3b965a559542
(maybe because of the formatting)

Reply:
Well, I think I agree. Let's keep it as is then.